examples/linear_model/plot_sparse_logistic_regression_20newsgroups.py

```diff
@@ -0,0 +1,114 @@
+"""
+=====================================================
+Multiclass logisitic regression on newgroups20
+=====================================================
+
+Comparison of multinomial logistic L1 vs one-versus-rest L1 logistic regression
+to classify documents from the newgroups20 dataset. Multinomial logistic
+regression yields more accurate results and is faster to train on the larger
+scale dataset.
+
+Here we use the l1 sparsity that trims the weights of no to informative
+features to zero. This is good if the goal is to extract the strongly
+discriminative vocabulary of each class. If the goal is to get the best
+predictive accuracy, it is better to use the non sparsity-inducing
+l2 penalty instead.
```

Comment:
I think you should also mention univariate feature selection as an alternative way to extract sparse discriminative vocabularies.

Maybe you could even extend the example by adding a pipeline of a sparse uni variate feature selection model +  l2 penalized logistic regression to showcase a classification model with similar sparsity level as the l1 penalized variant.